src/DetectionAlgorithms.jl

```diff
@@ -509,18 +496,18 @@ function calculateKNFST(K, labels)
     ### calculate weights of orthonormal basis in kernel space
     centeredK = copy(K); # because we need original K later on again
     centerKernelMatrix(centeredK);
-    (basisvecsValues,basisvecs) = eig(centeredK);
+    (basisvecsValues,basisvecs) = eigen(centeredK);
     basisvecs = basisvecs[:,basisvecsValues .> 1e-12];
     basisvecsValues = basisvecsValues[basisvecsValues .> 1e-12];
-    basisvecsValues = diagm(1./sqrt.(basisvecsValues));
+    basisvecsValues = Matrix(Diagonal(1 ./ sqrt.(basisvecsValues)));
```

Comment:
You can omit the `Matrix` here entirely and save some allocations by using the sparse `Diagonal`. Another equivalent alternative would be 

basisvecsvValues = 1 ./ sqrt.(basisvecsValues)
basisvecs = basisvecs.*basisvecsValues'

matter of taste...

Reply:
done